File: core/model/pma.go
package model

type GetServersRequest struct {
	Name     string `form:"name" json:"name"`
	Page     int    `form:"page" json:"page"`
	PageSize int    `form:"page_size" json:"page_size"`
}

type GetServersResponse struct {
	Total   int              `json:"total"`
	Servers []*PmaServerInfo `json:"servers"`
}

type PmaServerInfo struct {
	Verbose string `json:"verbose"`
	Host    string `json:"host"`
	Port    string `json:"port"`
}

type AddServerRequest struct {
	Name    string `json:"name"`
	Verbose string `json:"verbose"`
	Host    string `json:"host"`
	Port    string `json:"port"`
}

type RemoveServerRequest struct {
	Name string `json:"name"`
	Host string `json:"host"`
	Port string `json:"port"`
}

File: frontend/src/entity/Pma.ts
export interface PmaComposeInfo {
  name: string;
  version: string;
  port: string;
  status: string;
  container?: string;
}

export interface PmaComposesResponse {
  total: number;
  composes: PmaComposeInfo[];
}

export interface PmaOperateRequest {
  name: string;
  operation: 'start' | 'stop' | 'restart';
}

export interface PmaSetPortRequest {
  name: string;
  port: string;
}

export interface PmaServerInfo {
  verbose: string;
  host: string;
  port: string;
}

export interface PmaGetServersResponse {
  total: number;
  servers: PmaServerInfo[];
}

export interface PmaGetServersParams {
  // phpMyAdmin compose 名称
  name: string;
  page: number;
  page_size: number;
}

export interface PmaAddOrUpdateServerRequest {
  name: string;
  verbose: string;
  host: string;
  port: string;
}

export interface PmaRemoveServerRequest {
  name: string;
  host: string;
  port: string;
}
